### PR TITLE
add results to viewer: smarter defaults when overwriting

### PR DESCRIPTION
**Pull request by kecnry**
### Description

This pull request addresses an issues discovered in the review of #1514 with wider implications.  When adding results from a plugin to a viewer that will overwrite an existing data entry, selecting to not show in the viewer raised a traceback error.  This traceback alone could be fixed and could result in "unloading" the data entry from the viewer, but this then will result in issues if that data entry was the reference data (which usually isn't allowed, but #1514 provides an exception where the original reference data was created by a plugin and allowed to be overwritten).

**Previous behavior**:

https://user-images.githubusercontent.com/877591/182617186-9a769e00-f03a-42bd-887d-ddf5546088f7.mov

**Proposed new behavior**:

https://user-images.githubusercontent.com/877591/182617352-e1b1bb07-771c-4b2d-ba55-b590102da9a3.mov

and after iterating with @Jenneh, the label now shows as:

image

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [x] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`?
- [x] Is a milestone set?
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

**Comment by javerbukh** [comment]
I think as a temporary stand-in this makes sense. If others agree we should get this in soon to unblock #1514.

File: jdaviz/components/plugin_add_results.vue
Comment (pllim) [inline jdaviz/components/plugin_add_results.vue:28]: I find this disabled GUI confusing. Looks like you don't even need a switch anymore? Maybe I am missing something after being away for a week.

**Review by pllim — COMMENTED** [review]
What did the POs say about this?

File: jdaviz/components/plugin_add_results.vue
Comment (kecnry) [reply jdaviz/components/plugin_add_results.vue:28]: I tried mimicking the same layout without using a switch (the idea being to avoid a jarring shift in layout or styling), but ultimately found this to be simpler to implement.  Other ideas welcomed!

Comment (javerbukh) [reply jdaviz/components/plugin_add_results.vue:28]: Removing the switch entirely could also be a good temporary solution. I don't have strong feelings either way if we want to leave this up to the POs.

Comment (javerbukh) [reply jdaviz/components/plugin_add_results.vue:28]: I didn't see @kecnry's comment when I left my comment. Since this should hopefully be temporary, I'm ok with proceeding with this PR as is.

**Review by javerbukh — APPROVED** [review]

**Review by rosteen — APPROVED** [review]
I think this looks fine, we could maybe iterate on exactly what the UI looks like in this case in the future but I think this is good enough to fix a blocking bug. I tested with Gaussian Smooth and it works as expected.

**Pull request merged**
